db/view/view.cc

```diff
@@ -188,6 +189,10 @@ db::view::base_info_ptr view_info::make_base_dependent_view_info(const schema& b
     std::vector<column_id> base_regular_columns_in_view_pk;
     std::vector<column_id> base_static_columns_in_view_pk;
 
+    _has_same_partition_key = boost::equal(
+            base.partition_key_columns() | boost::adaptors::transformed(std::mem_fn(&column_definition::name)),
+            _schema.partition_key_columns() | boost::adaptors::transformed(std::mem_fn(&column_definition::name)));
```

Comment:
I think your optimization could be extended to cover more cases. Your optimization is applicable if and only if each partition in the base is paired with a single partition in the view table. For example, it's fine if the partition key columns are in a different order and/or are repeated in the view table. However, it's not fine if the partition key of the view table has some columns which are not partition key columns in the base table.

I think we already require that partition key of the view contains all partition key columns on the base table when creating the view (please check it).

Reply:
regarding the restriction on the view key, I think the restriction is that the view _primary key_ contains the base primary key. I don't think there is a restriction on the partition key.
For example, given a base table with PRIMARY KEY ((p1,p2),c), these are valid for the view primary key: (p1,p2,c), (c,p1,p2)
So I need to check in the code if the view partition key contains the base partition key.